pkg/logs/input/docker/convertor.go

```diff
@@ -0,0 +1,165 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+// +build docker
+
+package docker
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	iParser "github.com/DataDog/datadog-agent/pkg/logs/parser"
+	"regexp"
+)
+
+var (
+	delimiter = []byte{' '}
+	// 2019-05-29T09:26:32.155255473Z
+	timestampMatcher = regexp.MustCompile("\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{9}Z")
+	// Escaped CRLF, used for determine empty messages
+	windowsEOFMark     = []byte{'\\', 'r', '\\', 'n'}
+	carriageReturnMark = []byte{'\\', 'r'}
+	lineFeedMark       = []byte{'\\', 'n'}
+)
+
+const (
+	// [8ByteHeaderTimestamp Message]
+	numOfComponents = 2
+
+	// message indices
+	indexOfHeader  = 0
+	indexOfContent = 1
+
+	// 8Byte header indices
+	indexOfTimestamp   = 8
+	indexOfContentSize = 4
+	indexOfStatus      = 0
+)
+
+// Convertor specific for docker logs read from socket.
+type Convertor struct {
+	iParser.Convertor
+}
+
+// Convert converts the input message to Line, if it's partial msg, defaultPrefix will be
+// used as prefix.
+func (c *Convertor) Convert(msg []byte, defaultPrefix iParser.Prefix) *iParser.Line {
```

Comment:
I think it was nice to return errors with a specific message to understand why the parsing failed, it can help to debug when we are running into edge cases.